### ISSUE TITLE
Soft hyphenation in Weasyprint

### ISSUE THREAD

**Issue by jdus** [Kozea/WeasyPrint#176]
Hello,

I have been experimenting with hyphenation recently. Automatic hyphenation works like a charme. It's simply amazing. 

However, while manual hyphenation works, weasyprint never appends the actual hyphen character "-". It will break correctly, if needed, when encountering a shy but the trailing hyphen character is omitted.

I have tested it using this simple HTML file
```
<html>
<head>
</head>
<body>
ffffffffffffffffffffffffffffffffffffffffffffffffffffffg&shy;fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffg&shy;ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff&shy;fffffffffffffffffffffff&nsbp;fffffffffffffffffffffffffffffffffffff
</body>
</html>
```

Now, I have investigated and browsed the relevant parts in text.py, but it appears as if weasyprint is only concerned with automatic hyphenation and does not do the manual hyphenation part at all.

Could anybody confirm this behaviour? Or give me a hint what I am doing wrong? If this is indeed a bug maybe somebody could point out the relevant parts of the code?

Thanks for any comments :)

**Comment by liZe** [comment]
Grouped in #301.

**Comment by liZe** [comment]
For the record: fixed before closing #301 by a random WeasyPrint version.
